src/devices/honeywell_cm921.c

```diff
@@ -0,0 +1,449 @@
+/** @file
+    Honeywell CM921 Thermostat
+
+    Copyright (C) 2020
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+/**
+Honeywell CM921 Thermostat.
+
+868Mhz FSK, PCM, Start/Stop bits, reversed, Manchester.
+*/
+
+#include <stdlib.h>
+#include <mongoose.h>
+
+#include "decoder.h"
+
+#define _DEBUG
+
+static char *hex(bitbuffer_t *b, int row) {
+    size_t lenB = b->bits_per_row[row]/8;
+    if (b->bits_per_row[row] % 8 != 0) lenB++;
+    char *r = malloc(lenB * 2 + 1);
+    cs_to_hex(r, b->bb[row], lenB);
```

Comment:
Use something like this instead:

```
    char    str_buf[1024];

    // Get time now

    // Make data string
    str_buf[0] = 0;
    for (unsigned n=0; n<out->length; n++) { sprintf(str_buf+n*2, "%02x", out->data[n]); }

```

We might create a helper for it one day.

Reply:
Yeah, helper would be nice, I think I had ended up with three different ways to convert to hex :) (Fixed now).